hbase-client/src/main/java/org/apache/hadoop/hbase/client/RawAsyncHBaseAdmin.java

```diff
@@ -3000,7 +3001,12 @@ private void getProcedureResult(long procId, CompletableFuture<Void> future, int
                 // If any region compaction state is MAJOR_AND_MINOR
                 // the table compaction state is MAJOR_AND_MINOR, too.
                 if (err2 != null) {
-                  future.completeExceptionally(unwrapCompletionException(err2));
+                  // api can throw NSRE when the region is not online(split, splitting, merge,
+                  // merging, etc). Those regions will not be filtered above so it's better to skip
+                  // regions which throw NSRE.
+                  if (!(err2.getCause() instanceof NotServingRegionException)) {
```

Comment:
We should not swallow `NotServingRegionException` here. With this patch, we don't complete future call and eventually we will see CompactionState.NONE in UI without proper info on how the request was completed.

We need better way. Just a question: waiting for region to come out of RIT seems valid right? Due to this issue, are we not able to compact the table ever?

Reply:
@virajjasani , we are completing the future in later part of the code. Previously we had an `isOffline` flag which was used to filter out the regions which were not online but now we are not updating that flag. This patch just provides a workaround to skip those regions while calculating the compactionState of a table. But yes, I agree there won't be proper info on how the request was completed.

> Just a question: waiting for region to come out of RIT seems valid right?
This issue isn't specifically only in RITscenarios. Even if the user purposely unassigns a region, the future is completed exceptionally. 

> Due to this issue, are we not able to compact the table ever?

Yes future is completed exceptionally and table will not be compacted ever (the patch doesn't aim to fix this, patch is only for `getCompactionState()` and not for `compact()`).

What do you say? How should we proceed?

Reply:
> the patch doesn't aim to fix this, patch is only for `getCompactionState()` and not for `compact()`

If we can fix `compact()` i.e asynchronous compaction to consider waiting for region to come online and then proceed for the table, that will be really great. If we can maintain offline regions of a table, and retry compacting them when they come online, that would be much better.
If region doesn't come online in some configurable time (~30/60 s), we can fail compaction with proper error message and even indicate how many regions of this table are offline for long time and hence, the request failed.

Reply:
yes, this approach makes more sense. I think it's better to add a new flag for tracking offline regions in RegionInfo as the old `isOffline()` serves other purpose.

Reply:
Not having all the context, but a quick question is: if the workaround is for the interpret in UI, we can parse exception in `table.jsp`? The `compact()` improvement can be separately tracked.

Reply:
ok @liuml07 , for this fix, will move the parsing to `table.jsp` . Also, the test case will not be needed so I'll remove that as well.

Reply:
I'll raise a seperate JIRA for `compact()` improvement.

Reply:
Sure, handling Exception in `table.jsp` is good idea for this Jira and `compact()` improvement can be in different Jira.

Reply:
@virajjasani I have updated the PR, resolving this review comment.